apps/ts.c

```diff
@@ -888,7 +889,8 @@ static TS_VERIFY_CTX *create_verify_ctx(const char *data, const char *digest,
         f = TS_VFY_VERSION | TS_VFY_SIGNER;
         if (data != NULL) {
             f |= TS_VFY_DATA;
-            if (TS_VERIFY_CTX_set_data(ctx, BIO_new_file(data, "rb")) == NULL)
+            out = BIO_new_file(data, "rb");
+            if (TS_VERIFY_CTX_set_data(ctx, out) == NULL)
                 goto err;
```

Comment:
Thank you for your detail comments. I have changed them accordingly.  Previous patches are generated directly by our tool. We will make some change and improvement to our tool, too.